File: python3/SimplifyPath.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        items = path.split("/")
        
        stack = []
        for item in items:
            if item == "" or item == ".":
                continue
            elif item == "..":
                if len(stack):
                    stack.pop()
            else:
                stack.append(item)
        
        output = ""
        for item in stack:
            output += "/"
            output += item
        
        return output if output != "" else "/"

File: c++/SimplifyPath.cpp
class Solution {
public:
    string simplifyPath(string path) {
        string delim = "/";
        string output;

        int start = 1;
        int end = path.find(delim, start);
        
        vector<string> pathStack;
        
        do {
            string substr = path.substr(start, end - start);
            if (substr.compare(".") && substr.compare("")) {
                if (!substr.compare("..")) {
                    if (pathStack.size())
                        pathStack.pop_back();
                } else {
                    pathStack.push_back(substr);
                }
            }
            start += substr.length();
            start += delim.length();
            end = path.find(delim, start);
        } while (start < path.length());
        
        for (string item : pathStack) {
            output += "/";
            output += item;
        }
        
        if (pathStack.size()) {
            return output;
        }
        return "/";
    }
};
